portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,36 @@
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/logins")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      doPost(request, response);
```

Comment:
Now I see what you talked about. Is "calling doPost from doGet" the recommended solution in project guide steps?

Reply:
In the tutorials they don't really say if it's okay to call "doPost from doGet", most examples they have only have one or the other. 

I've also read through some stackoverflow forums about the subject, and it sounds like there is no issue calling them like that but you shouldn't need to call them from each other.

I can do some more research and try and find another solution.

Reply:
Do you have to implement anything for doGet at all? You could just have it throw a `UnsupportedOperationException`. It just seems a little unnecessary/misleading to have the 2 methods that do the same thing.

Reply:
make sense. So we only send Post from html. Probably we only need to implement doPost for now?

Reply:
So after doing a bit of research, the doGet function is called whenever the webpage is requested so "/data". After the user logs in I redirect the page back to "/data" but because the DataServlet.java only had a doPost I need to also have a doGet aswell. Without the doGet I get an error 405 for not having a doGet function.

Reply:
That makes sense. It seems counterintuitive to have them do the same thing, since Get and Post are fundamentally different actions. If one of them doesn't actually need to get used, you could do something like the following.

```
@Override
  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
// Not implemented yet
throw new UnsupportedOperationException("doGet is not implemented yet");
}
```

Reply:
Done! I went ahead and added it to LoginServlet also.